extensions/wikia/AdEngine/js/template/bfaaDesktop.js

```diff
@@ -77,11 +78,42 @@ define('ext.wikia.adEngine.template.bfaaDesktop', [
 		}
 
 		if (uapVideo.isEnabled(params)) {
-			uapVideo.loadVideoAd(videoSettings);
+			var videoPromise = uapVideo.loadVideoAd(videoSettings);
+			videoPromise.then(function(vid) {
+				video = vid;
+				bfaaMagic(params);
+			});
+		}
+	}
+
+
+	function bfaaMagic(params) {
+		// TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)